### PR TITLE
add recommendation_rating component

### PR DESCRIPTION
**Pull request by kgaikwad**
System Detail Page needs to import this component -

```
import RecommendationRating from '../../Components/RecommendationRating/RecommendationRating';
<RecommendationRating system={  system } />
```
This system object must include `rating`.

Pending work:
- [x] on backend side work. 
PR raised - https://github.com/RedHatInsights/ros-backend/pull/37

### CODE REVIEW

**Comment by kgaikwad** [comment]
I have referred advisor rule rating while implementing this.  Also, I observed there is component called rule feedback. 
+ adding @karelhala  into loop for further review to get comments/suggestions.

Here, the rating is against system as per mocks up - https://marvelapp.com/prototype/9408d9j/screen/74816603

**Comment by karelhala** [comment]
@terezanovotna I checked the mocks and it looks slightly different than we use for system detail information. Might I suggest adjustments to it so we are consistent across entire platform?

**Comment by kgaikwad** [comment]
> @terezanovotna I checked the mocks and it looks slightly different than we use for system detail information. Might I suggest adjustments to it so we are consistent across entire platform?

@karelhala, I would like to know with these adjustments, do I need to change anything for ratings?
I am asking this because backend side RedHatInsights/ros-backend#37 is ready to merge so if any change will be required on backend side with these adjustments, I will ask team to hold on for some time.

**Comment by terezanovotna** [comment]
@karelhala can you be specific about the differences? I believe this looks as a detail page in the platform

File: src/Components/RecommendationRating/RecommendationRating.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                body: JSON.stringify({
+                    inventory_id: system.inventory_id,  // eslint-disable-line camelcase
+                    rating: calculatedRating
+                })
+            };
+
+            await fetch(url, requestOptions);
+            setRating(calculatedRating);
+            setSubmitted(true);
+            setTimeout(() => setThankYou(''), 3000);
```
[the remark this answers is not in the document]
Comment (kgaikwad) [reply src/Components/RecommendationRating/RecommendationRating.js:34]: >  Also it has a bug that if user clicks multiple times on thumbs up/down the message is shwon only once.

Yes, I have noticed this behaviour that it only shows 'thank you' on first submit.

File: src/Components/RecommendationRating/RecommendationRating.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import './RecommendationRating.scss';
+import { ROS_API_ROOT, RECOMMENDATION_RATING_API } from '../../constants';
+
+const RecommendationRating = ({ system }) => {
+    const [rating, setRating] = useState(system.rating);
+    const [submitted, setSubmitted] = useState(false);
+    const [thankYou, setThankYou] = useState('Thank you for your feedback!');
+    const updateRecommendationRating = async (newRating) => {
+        const calculatedRating = rating === newRating ? 0 : newRating;
+        let url = new URL(ROS_API_ROOT + RECOMMENDATION_RATING_API,  window.location.origin);
```
[the remark this answers is not in the document]
Comment (kgaikwad) [reply src/Components/RecommendationRating/RecommendationRating.js:17]: Without origin, I am getting below error -

```
asyncToGenerator.js:6 Uncaught (in promise) TypeError: Failed to construct 'URL': Invalid URL
    at _callee$ (RecommendationRating.js:17)
    at tryCatch (runtime.js:63)
    at Generator.invoke [as _invoke] (runtime.js:293)
    at Generator.next (runtime.js:118)
    at asyncGeneratorStep (asyncToGenerator.js:3)
    at _next (asyncToGenerator.js:25)
    at asyncToGenerator.js:32
    at new Promise (<anonymous>)
    at asyncToGenerator.js:21
    at updateRecommendationRating (RecommendationRating.js:15)
```

[the remark this answers is not in the document]
Comment (karelhala) [reply src/Components/RecommendationRating/RecommendationRating.js:17]: I wouldn't use the `URL` util at all and just constructed it as 
```JS
await fetch(`${ROS_API_ROOT}${RECOMMENDATION_RATING_API}`, requestOptions);
```

[the remark this answers is not in the document]
Comment (kgaikwad) [reply src/Components/RecommendationRating/RecommendationRating.js]: awesome! thank you @karelhala.
Removed URL object & constructed directly using constants as suggested.

File: src/Components/RecommendationRating/RecommendationRating.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { ThumbsDownIcon } from '@patternfly/react-icons';
+
+import './RecommendationRating.scss';
+import { ROS_API_ROOT, RECOMMENDATION_RATING_API } from '../../constants';
+
+const RecommendationRating = ({ system }) => {
+    const [rating, setRating] = useState(system.rating);
+    const [submitted, setSubmitted] = useState(false);
+    const updateRecommendationRating = async (newRating) => {
+        const calculatedRating = rating === newRating ? 0 : newRating;
```
Comment (karelhala) [inline src/Components/RecommendationRating/RecommendationRating.js:16]: I'm not a huge fan of random numbers in code, do you think you could create 3 constants instead using -1/0/1?
```JS
const NEGATIVE_FEEDBACK = -1;
const NEUTRAL_FEEDBACK = 0;
const POSITIVE_FEEDBACK = 1;
```

Comment (kgaikwad) [reply src/Components/RecommendationRating/RecommendationRating.js]: Done! I have added export statements above class component. Is it the right place to add?

**Review by karelhala — APPROVED** [review]
Looking good!

File: src/Components/RecommendationRating/RecommendationRating.js (hunk 3 above)
Comment (karelhala) [reply src/Components/RecommendationRating/RecommendationRating.js]: You can add some constants file to group all constants together, but I think it's good for now!

Comment (kgaikwad) [reply src/Components/RecommendationRating/RecommendationRating.js]: We already have one constants file so moved these constants into same file.

**Comment by kgaikwad** [comment]
Backend PR is merged & all checks for this PR are green so merging these changes.
Thank you @karelhala for review :ok_hand:

**Pull request merged**